cibuildwheel/__main__.py

```diff
@@ -61,25 +61,29 @@ def main():
     args = parser.parse_args()
 
     detect_obsolete_options()
-
```

Comment:
Maybe this empty should be kept for readability ? allows the whole logic block for platform detection to be on its own.

Reply:
Ok.  maybe this function should be renamed to `detect_and_fix_obsolete_environment_variables` or `fix_deprecated_environment_variables"?

Reply:
I like the second one `fix_deprecated_environment_variables`,  it's clear enough. No need to add `detect_and_`.

Reply:
this PR or another? it is not connected with main problem solved by this PR.

Reply:
Let's say another. Since @joerick approved, I'll merge this one.